modules/configuration/src/testFixtures/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -300,22 +360,22 @@ public long notificationCount() {
         return cfgRef.get();
     }
 
-    private static List<Field> getInjectConfigurationFields(Class<?> testClass) {
+    private static List<Field> getInjectConfigurationFields(Class<?> testClass, boolean forStatic) {
         return AnnotationSupport.findAnnotatedFields(
                 testClass,
                 InjectConfiguration.class,
                 field -> supportsAsConfigurationType(field.getType()),
                 HierarchyTraversalMode.TOP_DOWN
-        );
+        ).stream().filter(field -> Modifier.isStatic(field.getModifiers()) == forStatic).collect(Collectors.toList());
```

Comment:
Can we pass this condition to a predicate?

Reply:
This is a predicate, isn't it?

Reply:
I suggest adding it to line **field -> supportsAsConfigurationType(field.getType())**

Reply:
Got it, makes sense